### PR TITLE
Interface`icx_closeorder` RPC 

### PR DESCRIPTION
**Pull request by surangap**
#### What kind of PR is this?:

/kind feature 

#### What this PR does / why we need it:
This PR will interface `icx_closeorder` RPC to jellyfish
#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
depends on #407

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    const ordersAfterCloseOrder2: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.listOrders()
+    expect((ordersAfterCloseOrder2 as Record<string, ICXOrderInfo>)[createOrderTxId]).toBeUndefined()
+    expect((ordersAfterCloseOrder2 as Record<string, ICXOrderInfo>)[createOrder2TxId]).toBeUndefined()
+
+    // check accountDFI[idDFI] balance
+    const accountDFIBalance: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    expect(accountDFIBalance).toStrictEqual(accountDFIBeforeOrder)
+  })
+
+  // NOTE(surangap): check why this is failing
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts:112]: typo?

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    const ordersAfterClaim: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.listOrders()
+    const orderInfo: Record<string, ICXOrderInfo> = ordersAfterClaim as Record<string, ICXOrderInfo>
+    expect(orderInfo[createOrderTxId].tokenFrom).toStrictEqual(symbolDFI)
+    expect(orderInfo[createOrderTxId].chainTo).toStrictEqual('BTC')
+    expect(orderInfo[createOrderTxId].ownerAddress).toStrictEqual(accountDFI)
+    expect(orderInfo[createOrderTxId].receivePubkey).toStrictEqual('037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941')
+    expect(orderInfo[createOrderTxId].amountFrom).toStrictEqual(new BigNumber(15))
+    expect(orderInfo[createOrderTxId].amountToFill).toStrictEqual(new BigNumber(5))
+    expect(orderInfo[createOrderTxId].orderPrice).toStrictEqual(new BigNumber(0.01))
+    expect(orderInfo[createOrderTxId].amountToFillInToAsset).toStrictEqual(new BigNumber(0.0500000))
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts:288]: old assertions pattern

**Review by fuxingloh — APPROVED** [review]
looks good

**Review by canonbrother — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts (hunk 1 above)
Comment (surangap) [reply packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts:112]: yes

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts (hunk 2 above)
Comment (fuxingloh) [reply packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts:288]: update in next PR

Comment (surangap) [reply packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts:288]: yes, the test is about closing a partially settled order. Hence checking most of the amount details after partial settlement. I could remove one or two like the symbol.

**Pull request merged**